### PR TITLE
Fix filters bug

### PR DESCRIPTION
**Pull request by cwrw**
If a filter is present but empty in a query parameter, it is interpreted as an empty string. When splitting an empty string the result is an empty array. This means that we query an empty array instead of nil for filters which does not return the intended results. Instead we should return nil when the string is empty to query any filter and expect values that are nil to be returned.

```
[3] pry(main)> ''.split(',')
=> []
[4] pry(main)> Move.where(to_location_id: nil).count
   (8.9ms)  SELECT COUNT(*) FROM "moves" WHERE "moves"."to_location_id" IS NULL
=> 3320
[5] pry(main)> Move.where(to_location_id: []).count
   (0.6ms)  SELECT COUNT(*) FROM "moves" WHERE 1=0
=> 0
[6] pry(main)> Move.where(cancellation_reason: []).count
   (0.6ms)  SELECT COUNT(*) FROM "moves" WHERE 1=0
=> 0
[7] pry(main)> Move.where(cancellation_reason: nil).count
   (14.6ms)  SELECT COUNT(*) FROM "moves" WHERE "moves"."cancellation_reason" IS NULL
=> 28222
[8] pry(main)> Move.where(rejection_reason: []).count
   (0.5ms)  SELECT COUNT(*) FROM "moves" WHERE 1=0
=> 0
[9] pry(main)> Move.where(rejection_reason: nil).count
   (16.7ms)  SELECT COUNT(*) FROM "moves" WHERE "moves"."rejection_reason" IS NULL
=> 43062
```

### CODE REVIEW

File: app/services/moves/finder.rb [hunk 1]
```diff
@@ -59,7 +59,9 @@ def apply_filters(scope)
     end
 
     def split_params(name)
-      filter_params[name]&.split(',')
+      return if filter_params[name].blank?
```
Comment (smoothcontract) [inline app/services/moves/finder.rb:62]: I'd prefer that we have consistent behaviour with this across all of the endpoints that support a comma delimited list - so either we don't do this anywhere, or it should also be done in the corresponding `split_params` method in `Locations::Finder`, `Allocations::Finder` and `People::Finder`.

I'm also not entirely clear why the front end can't just omit empty filters in the request and then no changes would be necessary? 🤔

Comment (cwrw) [reply app/services/moves/finder.rb:62]: sure I've added them, we're going to test this, and if all is well I'll add all the necessary tests.

I think its mostly a bug/unintentional behaviour on our side, if I send a filter with an empty value I expect records to come back which have this value as empty rather than none.

File: app/services/allocations/finder.rb [hunk 2]
```diff
@@ -73,6 +73,8 @@ def apply_date_range_filters(scope)
     end
 
     def split_params(name)
+      return if filter_params[name].blank?
```
Comment (martyn-w) [inline app/services/allocations/finder.rb:76]: how about `filter_params[name].presence&.split(',')`

File: app/services/moves/finder.rb (hunk 1 above)
Comment (smoothcontract) [reply app/services/moves/finder.rb:62]: Okay, that's cool 👍

File: app/services/allocations/finder.rb (hunk 2 above)
Comment (smoothcontract) [reply app/services/allocations/finder.rb:76]: I think I prefer the code as @cwrw has it - it's easier to read and understand quickly

File: app/services/allocations/finder.rb [hunk 3]
```diff
@@ -73,6 +73,8 @@ def apply_date_range_filters(scope)
...
+
       filter_params[name]&.split(',')
```
Comment (smoothcontract) [inline app/services/allocations/finder.rb:78]: You could remove the `&` here now as it's guarded by the new line above.

```suggestion
      filter_params[name].split(',')
```

File: app/services/locations/finder.rb [hunk 4]
```diff
@@ -40,6 +40,8 @@ def apply_filters(scope)
     end
 
     def split_params(name)
+      return if filter_params[name].blank?
+
       filter_params[name]&.split(',')
```
Comment (smoothcontract) [inline app/services/locations/finder.rb:45]: Remove `&`

File: app/services/v2/people/finder.rb [hunk 5]
```diff
@@ -16,6 +16,8 @@ def call
     private
 
       def split_params(name)
+        return if filter_params[name].blank?
+
         filter_params[name]&.split(',')
```
Comment (smoothcontract) [inline app/services/v2/people/finder.rb:21]: Remove `&`

File: app/services/allocations/finder.rb [hunk 6]
```diff
@@ -73,7 +73,9 @@ def apply_date_range_filters(scope)
     end
 
     def split_params(name)
-      filter_params[name]&.split(',')
+      return if filter_params[name].blank?
+
+      filter_params[name].split(',')
```
Comment (cwrw) [inline app/services/allocations/finder.rb:78]: can't add tests for this as we will never have nil values for locations/status on allocations

File: app/services/locations/finder.rb [hunk 7]
```diff
@@ -40,7 +40,9 @@ def apply_filters(scope)
     end
 
     def split_params(name)
-      filter_params[name]&.split(',')
+      return if filter_params[name].blank?
```
Comment (cwrw) [inline app/services/locations/finder.rb:43]: can't add tests here either as those filters are all for ids

**Review by smoothcontract — APPROVED** [review]
LGTM 👍
